### DISCUSSION TITLE
Vulkan on box64 on Linux Deploy (chroot) / Termux (proot) on Android?

### DISCUSSION THREAD

**Discussion by raspiduino** [ptitSeb/box64#752] [Q&A]
Now most new Android devices support [Vulkan](https://www.vulkan.org/), a graphic (like OpenGL but much better) and computing API. Normally, application / native code on Android use Vulkan by loading `libvulkan.so.1` at runtime, either by requiring the library directly (as you can see using `ldd`) or load it dynamically using `dlopen`.

But here is the problem. Take example of a Linux x86_64 executable that use Vulkan, and I want to run that on box64. The app will ask for `libvulkan.so.1` and box64 will try to look for that. If I give it the correct `LD_LIBRARY_PATH` it will find the shared library at `/system/lib64/libvulkan.so.1`.

But, box64 and my program is linked with glibc (so far I have never successes in compiling box64 as native Android binary), while `libvulkan.so.1` on Android (and other stubs it loads when the library initialize) is linked with Android's Bionic libc. These two things are not compatible with each other, so I cannot load 2 libc at the same time (tried that, got some signal and it crash).

What should I do to load Vulkan? I'm currently implementing a client/server bridge, but it is not a good idea for sure, and it's also hard. Thank you!

**Comment by ptitSeb** [comment]
This is a question for Termux team, not box64. Box64 is a linux app running inside Termux on Android.

For now, there is no solution to build box64 directly for Android (and even if it was doable, android is not linux so andoid/box64 would be usefull only to run x86/android program).
